In [1]:
from google.colab import drive
drive.mount('/content/drive')


Mounted at /content/drive


In [2]:
import os
for item in os.listdir('/content/drive/MyDrive'):
    print(repr(item))

'How to get started with Drive.pdf'
'ATTENDANCE LIST FOR 2015 ZONAL YOUTH  CONGRESS .xlsx'
'olusegun doc1.jpeg'
'Statement of Account (from 01-Apr-2021 to 30-Apr-2021).pdf'
'Statement of Account (from 01-May-2021 to 31-May-2021).pdf'
'Statement of Account (from 01-Jun-2021 to 30-Jun-2021).pdf'
'Uche O..pdf'
'Classroom'
'Bro uche ogbonanya.docx'
'CPC'
'CPC PROMOTIONAL'
'_PORTAL_MPC_DO_NOT_DELETE_'
'Colab Notebooks'
'WhatsApp Chat with Belove (1)'
'WhatsApp Chat with Belove'
'Copy of assignment_on_numpy.ipynb'
'Document from OGBONNAYA UCHECHUKWU. (2)'
'Document from OGBONNAYA UCHECHUKWU. (1)'
'Document from OGBONNAYA UCHECHUKWU.'
'waec_pass_dataset-1.csv'
'WAEC_Pass_Likelihood_Project.zip'
'2026-08-02 14-56-35.mp4'
'Data Science_copy.pdf'
'2026-08-07 14-09-37.zip'
'Waec_Pass_Likelihood_Project_folder'
'Demo For Capstone Project (1)'
'Demo For Capstone Project'
'Meridian Retail Group - Customer Churn Capstone '
'Copy of phase_2_Data_Acquisition.ipynb'


In [3]:
import os

meridian_folder = next(
    item for item in os.listdir('/content/drive/MyDrive')
    if item.startswith('Meridian Retail Group')
)

print("FOUND:", repr(meridian_folder))

FOUND: 'Meridian Retail Group - Customer Churn Capstone '


In [4]:
meridian_path = os.path.join(
    '/content/drive/MyDrive',
    meridian_folder
)

print("Meridian path:")
print(meridian_path)

print("\nContents:")
print(os.listdir(meridian_path))

Meridian path:
/content/drive/MyDrive/Meridian Retail Group - Customer Churn Capstone 

Contents:
['Business Understanding ', 'Data Collection & Preparation', 'SQL Analysis', 'Data']


In [5]:
import os

for item in os.listdir('/content/drive/MyDrive'):
    print(repr(item))

'How to get started with Drive.pdf'
'ATTENDANCE LIST FOR 2015 ZONAL YOUTH  CONGRESS .xlsx'
'olusegun doc1.jpeg'
'Statement of Account (from 01-Apr-2021 to 30-Apr-2021).pdf'
'Statement of Account (from 01-May-2021 to 31-May-2021).pdf'
'Statement of Account (from 01-Jun-2021 to 30-Jun-2021).pdf'
'Uche O..pdf'
'Classroom'
'Bro uche ogbonanya.docx'
'CPC'
'CPC PROMOTIONAL'
'_PORTAL_MPC_DO_NOT_DELETE_'
'Colab Notebooks'
'WhatsApp Chat with Belove (1)'
'WhatsApp Chat with Belove'
'Copy of assignment_on_numpy.ipynb'
'Document from OGBONNAYA UCHECHUKWU. (2)'
'Document from OGBONNAYA UCHECHUKWU. (1)'
'Document from OGBONNAYA UCHECHUKWU.'
'waec_pass_dataset-1.csv'
'WAEC_Pass_Likelihood_Project.zip'
'2026-08-02 14-56-35.mp4'
'Data Science_copy.pdf'
'2026-08-07 14-09-37.zip'
'Waec_Pass_Likelihood_Project_folder'
'Demo For Capstone Project (1)'
'Demo For Capstone Project'
'Meridian Retail Group - Customer Churn Capstone '
'Copy of phase_2_Data_Acquisition.ipynb'


In [6]:
import os

items = os.listdir('/content/drive/MyDrive')

meridian_folder = [item for item in items if item.startswith('Meridian Retail Group')][0]

print("Exact folder name found:")
print(repr(meridian_folder))

Exact folder name found:
'Meridian Retail Group - Customer Churn Capstone '


In [7]:
meridian_path = os.path.join(
    '/content/drive/MyDrive',
    meridian_folder
)

print(meridian_path)
print(os.listdir(meridian_path))

/content/drive/MyDrive/Meridian Retail Group - Customer Churn Capstone 
['Business Understanding ', 'Data Collection & Preparation', 'SQL Analysis', 'Data']


In [8]:
data_path = os.path.join(meridian_path, 'Data')
print(os.listdir(data_path))

['Raw Data']


In [9]:
raw_data_path = os.path.join(data_path, 'Raw Data')
print(os.listdir(raw_data_path))

['orders.csv', 'customers.csv', 'support_tickets.csv', 'products.csv', 'order_items.csv']


In [10]:
import os

meridian_path = os.path.join(
    '/content/drive/MyDrive',
    meridian_folder
)

print("Meridian folder:")
print(meridian_path)

print("\nFolders/files inside it:")
print(os.listdir(meridian_path))

Meridian folder:
/content/drive/MyDrive/Meridian Retail Group - Customer Churn Capstone 

Folders/files inside it:
['Business Understanding ', 'Data Collection & Preparation', 'SQL Analysis', 'Data']


In [11]:
data_path = os.path.join(meridian_path, 'Data')

print(os.listdir(data_path))

['Raw Data']


In [12]:
raw_data_path = os.path.join(data_path, 'Raw Data')

print("Raw Data folder:")
print(raw_data_path)

print("\nFiles inside Raw Data:")
print(os.listdir(raw_data_path))

Raw Data folder:
/content/drive/MyDrive/Meridian Retail Group - Customer Churn Capstone /Data/Raw Data

Files inside Raw Data:
['orders.csv', 'customers.csv', 'support_tickets.csv', 'products.csv', 'order_items.csv']


In [13]:
import os
import pandas as pd

# Your Raw Data folder
raw_data_path = os.path.join(data_path, 'Raw Data')

# Show the files available
print("Files in Raw Data:")
print(os.listdir(raw_data_path))

Files in Raw Data:
['orders.csv', 'customers.csv', 'support_tickets.csv', 'products.csv', 'order_items.csv']


In [14]:
# Load the CSV files

customers = pd.read_csv(
    os.path.join(raw_data_path, 'customers.csv')
)

products = pd.read_csv(
    os.path.join(raw_data_path, 'products.csv')
)

orders = pd.read_csv(
    os.path.join(raw_data_path, 'orders.csv')
)

order_items = pd.read_csv(
    os.path.join(raw_data_path, 'order_items.csv')
)

support_tickets = pd.read_csv(
    os.path.join(raw_data_path, 'support_tickets.csv')
)

print("All files loaded successfully!")

All files loaded successfully!


In [15]:
print("Customers:", customers.shape)
print("Products:", products.shape)
print("Orders:", orders.shape)
print("Order Items:", order_items.shape)
print("Support Tickets:", support_tickets.shape)

Customers: (6090, 14)
Products: (220, 9)
Orders: (28297, 9)
Order Items: (73505, 5)
Support Tickets: (5200, 8)


In [16]:
display(customers.head())
display(products.head())
display(orders.head())
display(order_items.head())
display(support_tickets.head())

,customer_id,first_name,last_name,email,phone,gender,date_of_birth,signup_date,state,city,membership_tier,preferred_channel,marketing_opt_in,referral_source
0,CUST002565,Karen,Castro,karen.castro168@gmail.com,784-499-9916x95359,F,1957-07-03,2026-05-28,FL,Jeffreybury,SILVER,Online,no,Email Campaign
1,CUST004275,Tonya,Webster,tonya.webster780@gmail.com,3985078721,M,1966-12-07,2022-12-23,Massachusetts,Michaelfurt,SILVER,In-Store,no,Search Engine
2,CUST003045,Alexis,Thomas,alexis.thomas522@gmail.com,5624741570,Other,1998-06-14,2022-07-10,Wisconsin,East Mark,Bronze,In-Store,1,TV Ad
3,CUST005168,Danielle,Goodman,danielle.goodman202@gmail.com,505.988.6915x03091,M,1975-08-30,11/03/2022,in,Ramosberg,BRONZE,Online,Y,Friend Referral
4,CUST003986,Daniel,Bowman,daniel.bowman295@yahoo.com,(988)305-0745x11524,F,1999-09-15,2023-07-05,TX,Walkerside,SILVER,Online,Y,NaN


,product_id,product_name,category,sub_category,brand,unit_cost,unit_price,launch_date,is_active
0,PROD1000,Cadence Wellness Classic,Beauty & Personal Care,Wellness,Cadence,152.32,415.24,2023-09-08,Yes
1,PROD1001,Northfield Cookware Pro,Home & Kitchen,Cookware,Northfield,293.87,645.92,2025-11-10,Yes
2,PROD1002,Pinecrest Accessories Essential,Apparel,Accessories,Pinecrest,65.78,$100.91,2026-03-13,Yes
3,PROD1003,Vertex Camera Essential,Electronics,Cameras,Vertex,27.00,70.18,2023-02-06,Yes
4,PROD1004,Lumen Wellness Plus,Beauty & Personal Care,Wellness,Lumen,242.04,571.72,2025-04-04,Yes


,order_id,customer_id,order_date,channel,order_status,payment_method,shipping_cost,discount_pct,total_amount
0,ORD118371,CUST005211,2026-02-14,Online,Completed,Gift Card,0.00,5,1098.00
1,ORD103955,CUST004689,2021-06-07,Mobile App,Completed,PayPal,0.00,0,349.47
2,ORD102242,CUST006035,2023-06-01,Online,Completed,Debit Card,12.99,15,981.00
3,ORD125665,CUST002265,2026-04-05,Mobile App,Completed,Debit Card,0.00,0,120.61
4,ORD108154,CUST000471,2022-06-28,Mobile App,Completed,Debit Card,5.99,0,225.89


,order_item_id,order_id,product_id,quantity,unit_price_at_purchase
0,OI562911,ORD116225,PROD1021,3,45.21
1,OI506869,ORD102266,PROD1124,2,982.99
2,OI529520,ORD100873,PROD1129,1,326.95
3,OI545253,ORD126873,PROD1097,2,247.92
4,OI526160,ORD105507,PROD1025,1,940.93


,ticket_id,customer_id,ticket_date,issue_category,channel,resolution_time_hours,satisfaction_score,ticket_text
0,TIX100000,CUST004178,2026-04-02,Delivery Delay,Email,9.8,5.0,My order ORD107221 still hasn't arrived and it...
1,TIX100001,CUST003557,2025-08-29,Wrong Item Shipped,Phone,14.2,4.0,When will my refund for the returned item be p...
2,TIX100002,CUST002786,2024-10-31,Damaged Item,Social Media,2.1,4.0,Can you tell me the warranty period on this pr...
3,TIX100003,CUST003978,2026-06-24,Billing Issue,Social Media,16.2,5.0,"I can't log into my account, it says my passwo..."
4,TIX100004,CUST001008,2025-09-13,Damaged Item,Phone,8.8,5.0,"I can't log into my account, it says my passwo..."


In [17]:

print("CUSTOMERS")
print(customers.info())

print("\nPRODUCTS")
print(products.info())

print("\nORDERS")
print(orders.info())

print("\nORDER ITEMS")
print(order_items.info())

print("\nSUPPORT TICKETS")
print(support_tickets.info())

CUSTOMERS
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6090 entries, 0 to 6089
Data columns (total 14 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   customer_id        6090 non-null   object
 1   first_name         6090 non-null   object
 2   last_name          6090 non-null   object
 3   email              5848 non-null   object
 4   phone              5667 non-null   object
 5   gender             6090 non-null   object
 6   date_of_birth      5909 non-null   object
 7   signup_date        6090 non-null   object
 8   state              6090 non-null   object
 9   city               5969 non-null   object
 10  membership_tier    6090 non-null   object
 11  preferred_channel  6090 non-null   object
 12  marketing_opt_in   4899 non-null   object
 13  referral_source    4385 non-null   object
dtypes: object(14)
memory usage: 666.2+ KB
None

PRODUCTS
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 220 entries, 0 to 219
D

In [18]:
print("Customers shape:", customers.shape)
print("Products shape:", products.shape)
print("Orders shape:", orders.shape)
print("Order Items shape:", order_items.shape)
print("Support Tickets shape:", support_tickets.shape)

Customers shape: (6090, 14)
Products shape: (220, 9)
Orders shape: (28297, 9)
Order Items shape: (73505, 5)
Support Tickets shape: (5200, 8)


In [19]:
# PHASE 2 — DATA ACQUISITION
# Dataset dimensions and data types
# ==========================================

datasets = {
    "Customers": customers,
    "Products": products,
    "Orders": orders,
    "Order Items": order_items,
    "Support Tickets": support_tickets
}

for name, df in datasets.items():
    print("=" * 60)
    print(f"{name.upper()}")
    print("=" * 60)

    print(f"Rows    : {df.shape[0]}")
    print(f"Columns : {df.shape[1]}")

    print("\nColumn Data Types:")
    print(df.dtypes)

    print("\n")

CUSTOMERS
Rows    : 6090
Columns : 14

Column Data Types:
customer_id          object
first_name           object
last_name            object
email                object
phone                object
gender               object
date_of_birth        object
signup_date          object
state                object
city                 object
membership_tier      object
preferred_channel    object
marketing_opt_in     object
referral_source      object
dtype: object


PRODUCTS
Rows    : 220
Columns : 9

Column Data Types:
product_id       object
product_name     object
category         object
sub_category     object
brand            object
unit_cost       float64
unit_price       object
launch_date      object
is_active        object
dtype: object


ORDERS
Rows    : 28297
Columns : 9

Column Data Types:
order_id           object
customer_id        object
order_date         object
channel            object
order_status       object
payment_method     object
shipping_cost     float64
discount_p

In [20]:
# Create a clean Phase 2 summary table

phase2_summary = pd.DataFrame({
    "Dataset": [
        "Customers",
        "Products",
        "Orders",
        "Order Items",
        "Support Tickets"
    ],
    "Rows": [
        customers.shape[0],
        products.shape[0],
        orders.shape[0],
        order_items.shape[0],
        support_tickets.shape[0]
    ],
    "Columns": [
        customers.shape[1],
        products.shape[1],
        orders.shape[1],
        order_items.shape[1],
        support_tickets.shape[1]
    ]
})

display(phase2_summary)

,Dataset,Rows,Columns
0,Customers,6090,14
1,Products,220,9
2,Orders,28297,9
3,Order Items,73505,5
4,Support Tickets,5200,8


In [21]:
for name, df in datasets.items():
    print(f"\n{name} columns:")
    print(list(df.columns))



Customers columns:
['customer_id', 'first_name', 'last_name', 'email', 'phone', 'gender', 'date_of_birth', 'signup_date', 'state', 'city', 'membership_tier', 'preferred_channel', 'marketing_opt_in', 'referral_source']

Products columns:
['product_id', 'product_name', 'category', 'sub_category', 'brand', 'unit_cost', 'unit_price', 'launch_date', 'is_active']

Orders columns:
['order_id', 'customer_id', 'order_date', 'channel', 'order_status', 'payment_method', 'shipping_cost', 'discount_pct', 'total_amount']

Order Items columns:
['order_item_id', 'order_id', 'product_id', 'quantity', 'unit_price_at_purchase']

Support Tickets columns:
['ticket_id', 'customer_id', 'ticket_date', 'issue_category', 'channel', 'resolution_time_hours', 'satisfaction_score', 'ticket_text']


 Phase 2 — Data Acquisition

 2.1 Data Acquisition and Initial Inspection

 The Meridian Retail Group customer churn project uses five operational datasets: Customers, Products, Orders, Order Items, and Support Tickets. The datasets were loaded from the project's Raw Data folder stored in Google Drive and imported into the Google Colab environment using Python and Pandas.

 An initial inspection was performed to determine the number of records and variables in each dataset and to identify the data types of the available columns. The results of this inspection are presented below.

# |index|Dataset|Rows|Columns|
# |---|---|---|---|
# |0|Customers|6090|14|
# |1|Products|220|9|
# |2|Orders|28297|9|
# |3|Order Items|73505|5|
# |4|Support Tickets|5200|8|

 The data-type inspection was performed using Pandas to identify numerical, categorical, date/time and other fields. This initial assessment provides the foundation for subsequent data-quality checks and preparation.

2.2 Entity-Relationship Structure

The five datasets represent interconnected business entities. Customers are connected to Orders through the customer identifier. Orders are connected to Order Items through the order identifier, while Order Items connect to Products through the product identifier. Customers are also connected to Support Tickets through the customer identifier.

This structure allows the analysis to combine customer information, purchasing behaviour, product information and customer-service interactions when investigating customer churn.

In [ ]:
┌──────────────────┐
                    │    CUSTOMERS     │
                    │------------------│
                    │ Customer_ID (PK) │
                    └────────┬─────────┘
                             │
                    1        │        Many
                             ▼
                    ┌──────────────────┐
                    │      ORDERS      │
                    │------------------│
                    │ Order_ID (PK)    │
                    │ Customer_ID (FK) │
                    └────────┬─────────┘
                             │
                    1        │        Many
                             ▼
                    ┌──────────────────┐
                    │   ORDER_ITEMS    │
                    │------------------│
                    │ Order_ID (FK)    │
                    │ Product_ID (FK)  │
                    │ Quantity         │
                    └────────┬─────────┘
                             │
                    Many     │        1
                             ▼
                    ┌──────────────────┐
                    │     PRODUCTS     │
                    │------------------│
                    │ Product_ID (PK)  │
                    └──────────────────┘


                    ┌──────────────────┐
                    │    CUSTOMERS     │
                    └────────┬─────────┘
                             │
                    1        │        Many
                             ▼
                    ┌────────────────────┐
                    │  SUPPORT_TICKETS   │
                    │--------------------│
                    │ Ticket_ID (PK)     │
                    │ Customer_ID (FK)   │
                    └────────────────────┘

In [ ]:
2.3 Business Origin of the Datasets

Dataset| Plausible Operational Source| Business Purpose
Customers| CRM / Customer Master System| Stores customer account and profile information
Products| Product / Inventory Management System| Stores information about products sold by Meridian
Orders| E-commerce, POS or Order Management System| Records customer orders and transactions
Order Items| Order Management / Transaction System| Records the individual products and quantities contained in each order
Support Tickets| Customer Service / Help-Desk / CRM System| Records customer enquiries, complaints and support interactions

These datasets were acquired as raw operational data and will subsequently be examined for missing values, duplicate records, inconsistent formats, incorrect data types and other quality issues during the data preparation stage.

2.3 Business Origin of the Datasets

Dataset| Plausible Operational Source| Business Purpose
Customers| CRM / Customer Master System| Stores customer account and profile information
Products| Product / Inventory Management System| Stores information about products sold by Meridian
Orders| E-commerce, POS or Order Management System| Records customer orders and transactions
Order Items| Order Management / Transaction System| Records the individual products and quantities contained in each order
Support Tickets| Customer Service / Help-Desk / CRM System| Records customer enquiries, complaints and support interactions

These datasets were acquired as raw operational data and will subsequently be examined for missing values, duplicate records, inconsistent formats, incorrect data types and other quality issues during the data preparation stage.

In [ ]:
# CUSTOMERS COLUMNS
Customer_ID          ← Primary Key
First_Name
Last_Name
Email
Phone
Gender
Date_of_Birth
Sign_Up_Date
State
City
Membership_Type
Preferred_Channel
Marketing_Opt_In
Referrer_Source

## PRODUCT COLUMNS
Product_ID           ← Primary Key
Product_Name
Category
Sub_Category
Brand
Unit_Cost
Unit_Price
Launch_Date
Is_Active

### ORDER COLUMNS
Order_ID             ← Primary Key
Customer_ID          ← Foreign Key
Order_Date
Channel
Order_Status
Payment_Method
Shipping_Cost
Discount_PCT

#### ORDER ITEMS COLUMNS
Order_Item_ID        ← Primary Key
Order_ID             ← Foreign Key
Product_ID           ← Foreign Key
Quantity
Unit_Price_at_Purchase

##### SUPPORT TICKET COLUMNS
Ticket_ID            ← Primary Key
Customer_ID          ← Foreign Key
Ticket_Date
Issue_Category
Channel
Resolution_Time_Hours
Satisfaction_Score
Ticket_Text